tokio/src/sync/batch_semaphore.rs

```diff
@@ -185,6 +184,11 @@ impl Semaphore {
         }
     }
 
+    /// Returns true if the semaphore is closed
+    pub(crate) fn is_closed(&self) -> bool {
+        self.permits.load(Acquire) & Self::CLOSED == Self::CLOSED
```

Comment:
very, very minor nit: here, we use `curr & Self::CLOSED == Self::CLOSED`, but in `try_acquire`, we use `curr & Self::CLOSED > 0`. these ought to be equivalent in practice, but i've a slight preference for using one consistently...

Reply:
I updated the other one.